lesson22/src/js/main.js

```diff
@@ -202,7 +202,7 @@ const sortTableBody = usersData => {
 
             const currentState = e.target.getAttribute("data-state");
             const nextState  = changeState(currentState);
-            const nextButton = document.querySelector(`[data-state=${nextState}]`);
+            const nextButton = document.querySelector(`#js-${category}Buttons-Box [data-state=${nextState}]`);
 
             toggleHiddenClassForSortButton(e.target,nextButton);
             sortUsersData(category, usersData, nextState);
```

Comment:
Here is the part that was added in this issue. 
I've added an id to keep track of which columns have been added.

However, I feel that it could be made shorter.